agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -3531,16 +3442,71 @@ backhaul_manager::create_link_metrics_collector(const sLinkInterface &link_inter
 
 bool backhaul_manager::get_neighbor_links(
     const sMacAddr &neighbor_mac_filter,
-    std::map<std::string, std::vector<sMacAddr>> &neighbor_links_map)
+    std::map<sLinkInterface, std::vector<sLinkNeighbor>> &neighbor_links_map)
 {
-
     // TODO: Topology Database is required to implement this method.
     // Since we don't have it yet, method will return a single neighbor, the controller (which is
     // BTW incorrect for the second repeater if we have a chain topology, because then the
-    // controller is not a neighbor).
-    sMacAddr neighbor = network_utils::mac_from_string(controller_bridge_mac);
-    if ((neighbor_mac_filter == network_utils::ZERO_MAC) || (neighbor_mac_filter == neighbor)) {
-        neighbor_links_map[m_sConfig.wire_iface].push_back(neighbor);
+    // controller is not a neighbor) connected through the wired backhaul interface
+
+    sLinkInterface wired_interface;
+    wired_interface.iface_name = m_sConfig.wire_iface;
+
+    if (!get_media_type(wired_interface.iface_name, ieee1905_1::eMediaTypeGroup::IEEE_802_3,
+                        wired_interface.media_type)) {
+        LOG(ERROR) << "Unable to compute media type for interface " << wired_interface.iface_name;
+        return false;
+    }
+
+    if (!get_iface_mac(wired_interface.iface_name, wired_interface.iface_mac)) {
+        return false;
+    }
+
+    sLinkNeighbor neighbor;
+    neighbor.iface_mac = network_utils::mac_from_string(controller_bridge_mac);
+    neighbor.al_mac    = neighbor.iface_mac;
+    if ((neighbor_mac_filter == network_utils::ZERO_MAC) ||
+        (neighbor_mac_filter == neighbor.al_mac)) {
+        neighbor_links_map[wired_interface].push_back(neighbor);
+    }
+
+    // Also include a link for each associated client
+    for (const auto &slave : slaves_sockets) {
+
+        for (const auto &associated_clients_entry : slave->associated_clients_map) {
+            auto bssid              = associated_clients_entry.first;
+            auto associated_clients = associated_clients_entry.second;
+
+            sLinkInterface interface;
+            if (get_iface_name(bssid, interface.iface_name)) {
```

Comment:
(nitpick) The usual pattern is
```suggestion
            if (!get_iface_name(bssid, interface.iface_name)) {
                return false;
            }
            ... // with less indentation
```

Reply:
Done